docs/tutorials/craft-a-snap.rst

```diff
@@ -0,0 +1,549 @@
+from creating the build environment and the configuration file, to troubleshooting
+missing libraries and finding which interfaces to open.
+
+It should take 20 minutes to complete.
+
+You won't need to come prepared with deep knowledge of software packaging, but
+familiarity with Linux paradigms and terminal operations is required.
+
+Once you complete this tutorial, you'll have experience hand-crafting snaps that will be
+the basis for further work with creating snaps.
+
+
+Lesson plan
+-----------
+
+Put simply, this tutorial is a run through of the process of constructing a snap. We
+show you how to:
+
+- Start a snap project from scratch
+- State the project's essential information
+- Define the project source files and main program
+- Package the snap
+- Modify the build process to enhance the snap's contents
+- Share system resources with the snap
+
+
+What we'll work with
+--------------------
+
+The object of this tutorial is to package `pyfiglet
+<https://github.com/pwaller/pyfiglet>`_ as a personal test snap. It's a lightweight app
+for displaying text as ASCII art, and is simple to build and test.
+
+
+What you'll need
+----------------
+
+For this tutorial, you'll need:
+
+- A x64 system running Ubuntu 22.04 or Ubuntu 24.04
+- A local user with super user privileges
+- 20GB of free storage
+
+
+Install Snapcraft and LXD
+-------------------------
+
+Snapcraft is itself available as a snap. Let's begin by installing it. In a terminal,
+run:
+
+.. literalinclude:: code/craft-a-snap/task.yaml
+  :language: bash
+  :start-at: sudo snap install snapcraft --classic
+  :end-at: sudo snap install snapcraft --classic
+  :dedent: 2
+
+Next, let's add LXD to your system. It functions as the build provider and containerize
+the build environment.
+
+.. literalinclude:: code/craft-a-snap/task.yaml
+  :language: bash
+  :start-at: sudo snap install --channel 5.21/stable lxd
+  :end-at: sudo snap install --channel 5.21/stable lxd
+  :dedent: 2
+
+You also need to add your user account to the ``lxd`` group so you can access the tool's
+resources:
+
+.. literalinclude:: code/craft-a-snap/task.yaml
+  :language: bash
+  :start-at: sudo usermod -a -G lxd $USER
+  :end-at: sudo usermod -a -G lxd $USER
+  :dedent: 2
+
+Log out and back in to your account for the new group to become active. Then, check that
+you're a member of the group by running:
+
+.. literalinclude:: code/craft-a-snap/task.yaml
+  :language: bash
+  :start-at: groups $USER
+  :end-at: groups $USER
+  :dedent: 2
+
+Look for ``lxd`` in the output.
+
+Finally, initialise LXD with a lightweight, default configuration:
+
+.. literalinclude:: code/craft-a-snap/task.yaml
+  :language: bash
+  :start-at: lxd init --minimal
+  :end-at: lxd init --minimal
+  :dedent: 2
+
+
+Begin the project
+-----------------
+
+Every snap project resides in its own directory. Start by creating one in a development
+space on your system:
+
+.. code-block:: bash
+
+    mkdir ukuzama-pyfiglet
```

Comment:
We've circled back to the generic project name problem. I'd really like to avoid making the reader work with `$USER`, because it's unrealistic, and we instruct them to call the snap by name more than once.

I'm OK with choosing a different name.